miden-lib/asm/kernels/transaction/api.masm

```diff
@@ -664,15 +748,50 @@ end
 #! Returns the serial number of the note currently being processed.
 #! Panics if no note is not being processed.
 #!
-#! Inputs: []
+#! Inputs: [KERNEL_PROCEDURE_HASH]
 #! Outputs: [SERIAL_NUMBER]
 #!
 #! - SERIAL_NUMBER is the serial number of the note currently being processed.
 export.get_note_serial_number
+    # drop the procedure's hash
+    dropw
+    
     exec.note::get_serial_number
     # => [SERIAL_NUMBER]
 
     # drop the padding
     swapw dropw
     # => [SERIAL_NUMBER]
 end
+
+#! Executes a kernel procedure specified by its offset.
+#!
+#! Inputs: [procedure_offset, procedure_inputs, PAD(*)]
+#! Outputs: [procedure_outputs, PAD(*)]
```

Comment:
nit: maybe we should write `<procedure_inputs>` and `<procedure_outputs>` to make sure it is not confused with a single field element.

Also, maybe we should switch to doing something like `<pad(x)>` as well.

Reply:
I agree, `<procedure_inputs>` and `<procedure_outputs>` will make it more clear that it is not just one element. With that approach for `pad`s I would use just the `<pad>` without `(x)`: `<>` already show that there are several pads there.